### PR TITLE
Removing deprecated charset=utf-8 from Content-Type header

### PR DESCRIPTION
**Pull request by vikrama**
Addresses https://github.com/apollographql/apollo-android/issues/2882 . Keeping the string in api module as per Martin's suggestion.

### CODE REVIEW

File: apollo-http-cache/src/main/java/com/apollographql/apollo/cache/http/ResponseHeaderRecord.java [hunk 1]
```diff
@@ -272,7 +273,7 @@ private void writeCertList(BufferedSink sink, List<Certificate> certificates)
   Response response() {
     RequestBody body = null;
     if (HttpMethod.permitsRequestBody(requestMethod)) {
-      body = RequestBody.create(MediaType.parse("application/json; charset=utf-8"), "");
+      body = RequestBody.create(MediaType.parse(ApolloMediaType.APPLICATION_JSON_UTF8), "");
```
Comment (martinbonnin) [inline apollo-http-cache/src/main/java/com/apollographql/apollo/cache/http/ResponseHeaderRecord.java:276]: Any reason to keep the charset here?

Comment (vikrama) [reply apollo-http-cache/src/main/java/com/apollographql/apollo/cache/http/ResponseHeaderRecord.java]: Initial thought was that it won't affect the content-type header, but realized OkHttp uses the content type set in the body. Have removed the charset. Thanks.

**Comment by martinbonnin** [comment]
@vikrama Thanks for the contribution 🤗 

I moved everything to the `api.internal` package, I don't think this needs to be seen from consumers. We can always add it later if needed.

**Comment by vikrama** [comment]
Thx @martinbonnin for the review.

Not urgent, but trying to understand why my last build failed. Seems unrelated to my changes. Will help me be better informed while sending PRs in the future.

**Comment by martinbonnin** [comment]
>  trying to understand why my last build failed. Seems unrelated to my changes. Will help me be better informed while sending PRs in the future.

Are you refering about this build failure? https://github.com/apollographql/apollo-android/runs/1724495722

This is because we track changes to the public API with metalava https://android.googlesource.com/platform/tools/metalava/

This we we're notified any time the public API breaks. For this to work, we also have to track we new additions are made like is the case here.

To fix the build, you can run `./gradlew metalavaGenerateSignature`

**Pull request merged**